src/shared/sleep-config.c

```diff
@@ -200,6 +204,18 @@ int find_hibernate_location(char **device, char **type, uint64_t *size, uint64_t
 
         (void) fscanf(f, "%*s %*s %*s %*s %*s\n");
 
+        r = read_one_line_file("/sys/power/resume", &resume);
+        if (r < 0)
+                return log_debug_errno(r, "Error reading from /sys/power/resume: %m");
+
+        r = read_one_line_file("/sys/power/resume_offset", &resume_offset);
+        if (r < 0) {
+                if (r == -ENOENT)
+                        log_debug("Kernel does not support resume_offset; swap file offset detection will be skipped.");
+                else
+                        return log_debug_errno(r, "Error reading from /sys/power/resume_offset: %m");
+        }
+
```

Comment:
This copies a few lines of code from configure_hibernation(). Please don't copy copy, but instead turn it into a shared function — see the main comment above.